metrics-util/src/registry/mod.rs

```diff
@@ -456,6 +498,15 @@ mod tests {
         let value = registry.get_counter(&key).expect("failed to get entry");
         assert!(Arc::ptr_eq(&value, &uvalue));
 
+        registry.get_or_create_counter(&Key::from_name("baz"), |_| ());
+
+        let mut n = 0;
+        registry.retain_counters(|k, _| {
+            n += 1;
+            k.name().starts_with("foo")
+        });
+        assert_eq!(n, 2);
+
```

Comment:
I think you'd want to copy the assert from lines 512-513 and stick them after adding the `baz` counter, to show that we now have two counters in the registry. After that, do the same thing after the `retain_counters` call to show it goes back down to one counter after it deletes `baz`.